lib/tests/streamlit/runtime/state/session_state_test.py

```diff
@@ -555,6 +554,21 @@ def test_compact(self):
         assert self.session_state._new_session_state == {}
         assert self.session_state._new_widget_state == WStates()
 
+    # https://github.com/streamlit/streamlit/issues/7206
+    def test_ignore_key_error_within_compact_state(self):
+        # Arrange: Mock __getitem__ to throw a KeyError when it sees the key 'bad_key'
+        with patch.object(
```

Comment:
It would be a little better, though more effort, to instead set up the session state with a serialized widget with no metadata, to replicate the situation that causes the problem.

Reply:
Sounds good. Will fix that.